hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/operations/ReadManyOperation.java

```diff
@@ -79,11 +79,15 @@ public boolean shouldWait() {
             return false;
         }
 
-        if (ringbuffer.shouldWait(sequence)) {
+        if (sequence > ringbuffer.tailSequence() + 1
```

Comment:
Nice fix. I'd suggest just one improvement - let's add two methods:
- `RingbufferContainer.isStaleSequence(long sequence)`
-  `RingbufferContainer.isTooLargeSequence(long sequence)`
We can then reuse these methods in the operations and in `RingbufferContainer#checkBlockableReadSequence`.